src/Data/ExtractShardKey.ts

```diff
@@ -148,6 +157,56 @@ async function extractKeyFromInternalTx(tx: any): Promise<string> {
   return extractedKey
 }
 
+function toHexString(byteArray: Uint8Array): string {
+  return Array.from(byteArray, (byte) => {
+    return ('0' + (byte & 0xff).toString(16)).slice(-2)
+  }).join('')
+}
+
+export function getTxSenderAddress(
+  tx: TypedTransaction,
+  txid: string = undefined,
+  overrideSender: Address = undefined
+): GetTxSenderAddressResult {
+  try {
+    if (overrideSender != null) {
+      const res = { address: overrideSender, isValid: true, gasValid: true }
+      if (txid != null) {
+        txSenderCache.set(txid, res)
+      }
+      return res
+    }
+
+    if (txid != null) {
+      const cached = txSenderCache.get(txid)
+      if (cached != null) {
+        return cached
+      }
+    }
+
+    const rawTx = '0x' + toHexString(tx.serialize())
+    const { address, isValid, gasValid } = getSenderAddress(rawTx)
+    if (config.VERBOSE) console.log('Sender address retrieved from signed txn', address)
+    const res = { address: Address.fromString(address), isValid, gasValid }
+    if (txid != null) {
+      simpleTTL++
```

Comment:
should this be only incremented if txid is not in txSenderCache?
Also, if we are clearing the cache, shouldn't we set simpleTTL to 0?